cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1851,8 +1881,10 @@ const char* rpad_utf8_int32_utf8(gdv_int64 context, const char* text, gdv_int32
   } else {
     // case (return_length > text_char_count)
     // case where it needs to copy "fill_text" on the string right
-    char* ret =
-        reinterpret_cast<gdv_binary>(gdv_fn_context_arena_malloc(context, return_length));
+    gdv_int32 return_char_length = evaluate_return_char_length(
+        text_len, text_char_count, return_length, fill_text, fill_text_len);
+    char* ret = reinterpret_cast<gdv_binary>(
+        gdv_fn_context_arena_malloc(context, return_char_length));
```

Comment:
same comment as above. Shouldn't the while loop in 1899 use return_char_length instead of return length?